system-addon/content-src/components/Sections/Sections.jsx

```diff
@@ -108,42 +116,61 @@ export class Section extends React.PureComponent {
     return false;
   }
 
-  numberOfPlaceholders(items) {
-    if (items === 0) {
-      return CARDS_PER_ROW;
-    }
-    const remainder = items % CARDS_PER_ROW;
-    if (remainder === 0) {
-      return 0;
-    }
-    return CARDS_PER_ROW - remainder;
-  }
-
   render() {
     const {
       id, eventSource, title, icon, rows,
-      emptyState, dispatch, maxRows,
+      emptyState, dispatch, compactCards, maxRows,
       contextMenuOptions, initialized, disclaimer,
       pref, privacyNoticeURL, isFirst, isLast
     } = this.props;
-    const maxCards = CARDS_PER_ROW * maxRows;
+
+    const maxCardsPerRow = compactCards ? CARDS_PER_ROW_COMPACT_WIDE : CARDS_PER_ROW_DEFAULT;
+    const maxCards = maxCardsPerRow * maxRows;
+    const maxCardsOnNarrow = CARDS_PER_ROW_DEFAULT * maxRows;
 
     // Show topics only for top stories and if it's not initialized yet (so
     // content doesn't shift when it is loaded) or has loaded with topics
     const shouldShowTopics = (id === "topstories" &&
       (!this.props.topics || this.props.topics.length > 0));
 
     const realRows = rows.slice(0, maxCards);
-    const placeholders = this.numberOfPlaceholders(realRows.length);
 
     // The empty state should only be shown after we have initialized and there is no content.
     // Otherwise, we should show placeholders.
     const shouldShowEmptyState = initialized && !rows.length;
 
+    const cards = [];
+    if (!shouldShowEmptyState) {
+      for (let i = 0; i < maxCards; i++) {
+        const link = realRows[i];
+        // On narrow viewports, we only show 3 cards per row. We'll mark the rest as
+        // .hide-for-narrow to hide in CSS via @media query.
+        const className = (i >= maxCardsOnNarrow) ? "hide-for-narrow" : "";
```

Comment:
I see this is basically what we do for TopSites, so I guess it's fine. Just thinking about this a bit more, it's a bit strange for the API to have Card just use whatever className given instead of some boolean flag.

Reply:
Make a `hideForNarrow` prop instead? I guess that would keep it more strict/focused/scoped/whateveryoucallit?

Reply:
I wonder if `optional` would be too vague ;)